src/common/fs/FileUtils.cpp

```diff
@@ -134,6 +135,20 @@ std::string FileUtils::basename(const char *path) {
     return result[3].str();
 }
 
+std::tuple<double, double, double, double> FileUtils::detectFilesystemUsage(std::string data_path) {
+    const unsigned int GB = (1024 * 1024) * 1024;
+    struct statvfs buffer;
+    int ret = statvfs(data_path.c_str(), &buffer);
+
+    if (!ret) {
+        const double total = (double)(buffer.f_blocks * buffer.f_frsize) / GB;
+        const double available = (double)(buffer.f_bfree * buffer.f_frsize) / GB;
+        const double used = total - available;
+        const double usedPercentage = (double)(used / total) * (double)100;
```

Comment:
All you need to return with this function is _total_ and _available_. The caller is responsible to calculate _used_.
Similarly, you should return by bytes, instead of by GB mandatorily.

Reply:
thanks, I will do this as you advise.